runtime/stdlib/account.go

```diff
@@ -670,21 +678,117 @@ func newAccountKeysGetFunction(
 					inter,
 					locationRange,
 					accountKey,
-					// public keys are assumed to be already validated.
-					func(
-						_ *interpreter.Interpreter,
-						_ interpreter.LocationRange,
-						_ *interpreter.CompositeValue,
-					) error {
-						return nil
-					},
+					assumePublicKeyIsValid,
 				),
 			)
 		},
 		sema.AccountKeysTypeGetFunctionType,
 	)
 }
 
+func newAccountKeysForEachFunction(
+	gauge common.MemoryGauge,
+	provider AccountKeyProvider,
+	addressValue interpreter.AddressValue,
+) *interpreter.HostFunctionValue {
+	address := addressValue.ToAddress()
+	// the AccountKey in `forEachKey(_ f: ((AccountKey): Bool)): Void`
+	callbackTypeParams := []sema.Type{sema.AccountKeyType}
```

Comment:
can make this a module level variable

Reply:
hmm, this is only used in one place, don't think it's worth polluting the module namespace

Reply:
Avoids allocating a fresh slice, so maybe worth it